core/server/ntske.go

```diff
@@ -0,0 +1,97 @@
+package server
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"strconv"
+
+	"go.uber.org/zap"
+
+	"example.com/scion-time/net/ntske"
+)
+
+const defaultNtskePort int = 4460
+
+func runNTSKEServer(log *zap.Logger, listener net.Listener, localHost *net.UDPAddr, provider *ntske.Provider) {
+	for {
+		ke, err := ntske.NewListener(listener)
+		if err != nil {
+			log.Info("failed to accept client", zap.Error(err))
+			break
+		}
+
+		err = ke.Read()
+		if err != nil {
+			log.Info("failed to read key exchange", zap.Error(err))
+			return
+		}
+
+		err = ke.ExportKeys()
+		if err != nil {
+			log.Info("failed to export keys", zap.Error(err))
+			return
+		}
+
+		var msg ntske.ExchangeMsg
+		msg.AddRecord(ntske.NextProto{
+			NextProto: ntske.NTPv4,
+		})
+		msg.AddRecord(ntske.Algorithm{
+			Algo: []uint16{ntske.AES_SIV_CMAC_256},
+		})
+		msg.AddRecord(ntske.Server{
+			Addr: []byte(localHost.IP.String()),
+		})
+		msg.AddRecord(ntske.Port{
+			Port: uint16(localHost.Port),
+		})
+
+		var plaintextCookie ntske.ServerCookie
+		plaintextCookie.Algo = ntske.AES_SIV_CMAC_256
+		plaintextCookie.C2S = ke.Meta.C2sKey
+		plaintextCookie.S2C = ke.Meta.S2cKey
+		key := provider.Current()
+		for i := 0; i < 8; i++ {
+			encryptedCookie, err := plaintextCookie.EncryptWithNonce(key.Value, key.Id)
+			if err != nil {
+				log.Info("failed to encrypt cookie", zap.Error(err))
+				continue
```

Comment:
This only skips encoding and adding the current cookie (https://go.dev/ref/spec#Continue_statements). Is this the intended behavior or should we always provide 8 cookies and continue with the next key exchange instead. If it is intended behavior, I suggest adding a short comment.

Reply:
Yes, sounds reasonable.